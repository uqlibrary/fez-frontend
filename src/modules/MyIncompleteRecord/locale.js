import React from 'react';
import locale from 'locale/components';
import fileUploaderLocale from 'modules/SharedComponents/Toolbox/FileUploader/locale';

/*

NOTE:
- text can be either plain text, eg text: 'Some text to display' or
- text can be formatted HTML text, eg
    text: (<div>Click here to search google: <a href='google.com'>search google</a></div>)
IMPORTANT: if currently text contains placeholders, eg any characters in square brackets,
eg [noOfResults] it cannot be formatted with HTML tagsâ€™

- help objects have the following shape:
help: {
    title: 'About these metrics',
    text: (<div></div>),
    buttonLabel: 'CLOSE'
}
- text can be plain or formatted HTML component with links/tags/etc
- if help is not required, delete help: {} fully (including closing '},')

*/
/* eslint-disable max-len */
export default {
    title: 'Complete my work',
    help: {
        title: 'Add more information to an existing NTRO',
        text: (
            <p>
                For help, click{' '}
                <a
                    href="https://guides.library.uq.edu.au/for-researchers/uqespace-publications-datasets/add-missing-ntro-information"
                    target="_blank"
                >
                    here
                </a>
                .
            </p>
        ),
        buttonLabel: 'CLOSE',
    },
    submitButtonLabel: 'Complete work',
    cancelButtonLabel: 'Cancel and return to my incomplete works list',
    fields: {
        notes: {
            title: 'Notes',
            label: 'Notes for this work',
            placeholder: 'Add any other notes or comments about this work to send to the eSpace team.',
        },
        grants: {
            title: 'Grant information',
        },
        authors: {
            ...locale.components.authors,
            description: (
                <span>
                    For each author marked with a <span style={{ color: '#e60000', weight: 'bold' }}>red</span> prompt,
                    select the pencil edit icon to the right of the author name at Step 1, add the affiliation
                    information as at time of publication at Step 2, then click <b>CHANGE DETAILS</b>.
                </span>
            ),
            field: {
                ...locale.components.authors.field,
                form: {
                    locale: {
                        ...locale.components.authors.field.form.locale,
                        descriptionStep1: (
                            <div>
                                <span className="authorSteps" key="step-2">
                                    Step 2 of 2
                                </span>{' '}
                                - <b>Update the affiliation information</b>.
                            </div>
                        ),
                        addButton: 'Update author',
                    },
                },
                header: {
                    ...locale.components.authors.field.header,
                    locale: {
                        ...locale.components.authors.field.header.locale,
                        descriptionStep2: (
                            <div>
                                <span className="authorSteps" key="step-1">
                                    Step 1 of 2
                                </span>{' '}
                                - <b>Select an author</b>.
                            </div>
                        ),
                    },
                },
                row: {
                    ...locale.components.authors.field.row,
                    locale: {
                        ...locale.components.authors.field.row.locale,
                        selectHint: 'Select this author ([name]) to update their affiliation data.',
                        requiredLabel: "This author's data is incomplete.",
                    },
                },
            },
        },
        fileUpload: {
            title: 'Upload files',
            locale: {
                ...fileUploaderLocale,
                fileUploadRow: {},
                instructions: '',
                ntroSpecificInstructions: '',
                validation: {
                    ...fileUploaderLocale.validation,
                    ['sameFileNameWithDifferentExt']: '',
                },
            },
            text: (
                <div>
                    <span className="requiredField">
                        <label>&nbsp;</label>
                    </span>
                </div>
            ),
        },
    },
    successWorkflowConfirmation: {
        confirmationTitle: 'Your work has been updated',
        fileFailConfirmationAlert: {
            title: 'UPLOAD FAILED',
            message: 'File upload and/or notes post failed',
            type: 'warning',
        },
        cancelButtonLabel: 'Complete another work',
        confirmButtonLabel: 'Go to my dashboard',
    },
    prompt: {
        title: 'Missing data',
        message:
            'follow the red prompts to add missing information. You can also provide additional grant details or include notes to the eSpace team.',
        type: 'info_outline',
    },
    progressAlert: {
        type: 'info_outline',
        title: 'Saving',
        message: 'Updating work is in progress.',
        showLoader: true,
    },
    successAlert: {
        type: 'done',
        title: 'Success',
        message: 'New information has been saved successfully.',
    },
};
