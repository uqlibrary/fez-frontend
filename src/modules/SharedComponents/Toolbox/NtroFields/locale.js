import React from 'react';
import Typography from '@mui/material/Typography';

export default {
    contributionStatement: {
        title: 'Author/Creator research statement',
        fields: {
            scaleOfWork: {
                label: 'Scale/Significance of work',
                description: (
                    <span>
                        Select the option that best describes the significance of the work*
                        <span style={{ fontWeight: 700 }}> (not for public view)</span>
                    </span>
                ),
            },
            impactStatement: {
                label: (
                    <span>
                        Creator research statement*. Include Background, Contribution and Significance
                        <span style={{ fontWeight: 700 }}> (not for public view)</span>
                    </span>
                ),
                placeholder:
                    'Remember to include substantiation of your major or minor scale/significance claim above.',
            },
        },
        help: {
            title: 'Author/Creator research statement',
            text: (
                <React.Fragment>
                    <Typography component="h4" variant="h6">
                        Creator research statement
                    </Typography>
                    <p>
                        For more information about the research statement, click{' '}
                        <a
                            style={{ fontWeight: 700 }}
                            target="_blank"
                            href="https://guides.library.uq.edu.au/for-researchers/uqespace-publications-datasets/ntro-submission-requirements#s-lg-box-20836548"
                        >
                            here
                        </a>
                    </p>
                </React.Fragment>
            ),
            buttonLabel: 'CLOSE',
        },
    },
    metadata: {
        title: 'NTRO data',
        fields: {
            abstract: {
                label: (
                    <span>
                        Abstract/Description* <span style={{ fontWeight: 700 }}>(for public view)</span>
                    </span>
                ),
                placeholder: 'Enter a brief description of the work - 800 character limit is no longer applicable',
            },
            series: {
                floatingLabelText: 'Series',
                hintText: 'Enter the name of publication, performance, recording, or event series',
            },
            volume: {
                label: 'Volume',
            },
            issue: {
                label: 'Issue',
            },
            startPage: {
                label: 'Start page',
            },
            endPage: {
                label: 'End page',
            },
            extent: {
                label: 'Total pages/Extent',
                placeholder: 'Total pages, size, or duration',
            },
            physicalDescription: {
                label: 'Physical description',
                placeholder: 'e.g Building, Exhibit, Performance',
            },
            audienceSize: {
                label: 'Audience size',
            },
            peerReviewActivity: {
                label: 'Quality indicators',
            },
            notes: {
                label: 'Notes',
            },
            language: {
                label: 'Language',
                selectPrompt: 'Please select languages as required',
            },
        },
    },
    grantEditor: {
        title: 'Grant details',
    },
};
