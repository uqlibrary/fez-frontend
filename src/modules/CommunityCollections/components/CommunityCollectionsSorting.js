import React from 'react';
import PropTypes from 'prop-types';
import { locale } from 'locale';
import Select from '@material-ui/core/Select';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Grid from '@material-ui/core/Grid';

const CommunityCollectionsSorting = props => {
    const [sortBy, setSortBy] = React.useState(props.sortBy || locale.components.sorting.sortBy[0].value);
    const [sortDirection, setSortDirection] = React.useState(
        props.sortDirection || locale.components.sorting.sortDirection[0],
    );
    const [pageSize, setPageSize] = React.useState(
        props.pageSize || (props.pagingData && props.pagingData.per_page ? props.pagingData.per_page : 20),
    );

    React.useEffect(() => {
        if (sortBy !== props.sortBy) setSortBy(props.sortBy);
        if (sortDirection !== props.sortDirection) setSortDirection(props.sortDirection);
        if (pageSize !== props.pageSize) setPageSize(props.pageSize);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [props.sortBy, props.sortDirection, props.pageSize]);

    const pageSizeChanged = event => {
        setPageSize(event.target.value);
        props.onPageSizeChanged(event.target.value);
    };

    const orderDirectionsChanged = event => {
        setSortDirection(event.target.value);
        props.onSortByChanged(sortBy, event.target.value);
    };

    const sortByChanged = event => {
        setSortBy(event.target.value);
        props.onSortByChanged(event.target.value, sortDirection);
    };
    if (!props.pagingData || props.pagingData.total === 0 || !sortBy || !sortDirection || !pageSize) {
        return <span className="publicationsListSorting empty" />;
    }
    const txt = props.sortingData;
    const pageLength = txt.recordsPerPage;
    if (props.initPageLength && pageLength.indexOf(props.initPageLength) === -1) {
        pageLength.push(props.initPageLength);
        pageLength.sort((a, b) => a - b);
    }

    return (
        <Grid container spacing={2}>
            <Grid item xs={12} sm={4}>
                <FormControl fullWidth>
                    <InputLabel shrink>{txt.sortLabel}</InputLabel>
                    <Select id="sortBy" onChange={sortByChanged} value={sortBy} disabled={props.disabled}>
                        {txt.sortBy.map((item, index) => {
                            return (
                                <MenuItem key={index} value={item.value}>
                                    {item.label}
                                </MenuItem>
                            );
                        })}
                    </Select>
                </FormControl>
            </Grid>
            <Grid item xs={12} sm={4}>
                <FormControl fullWidth>
                    <InputLabel shrink>{txt.sortDirectionLabel}</InputLabel>
                    <Select
                        id="sortOrder"
                        onChange={orderDirectionsChanged}
                        value={sortDirection}
                        disabled={props.disabled}
                    >
                        {txt.sortDirection.map((item, index) => {
                            return (
                                <MenuItem key={index} value={item}>
                                    {item}
                                </MenuItem>
                            );
                        })}
                    </Select>
                </FormControl>
            </Grid>
            <Grid item xs={12} sm={4}>
                <FormControl fullWidth>
                    <InputLabel shrink>{txt.pageSize}</InputLabel>
                    <Select id="pageSize" value={pageSize} disabled={props.disabled} onChange={pageSizeChanged}>
                        {pageLength.map(number => {
                            return (
                                <MenuItem key={`records-per-page-${number}`} value={number}>
                                    {number}
                                </MenuItem>
                            );
                        })}
                    </Select>
                </FormControl>
            </Grid>
        </Grid>
    );
};

CommunityCollectionsSorting.propTypes = {
    bulkExportSize: PropTypes.number,
    canUseExport: PropTypes.bool,
    disabled: PropTypes.bool,
    initPageLength: PropTypes.number,
    onExportPublications: PropTypes.func,
    onPageSizeChanged: PropTypes.func,
    onSortByChanged: PropTypes.func,
    pageSize: PropTypes.number,
    pagingData: PropTypes.shape({
        from: PropTypes.number,
        to: PropTypes.number,
        total: PropTypes.number,
        per_page: PropTypes.number,
        current_page: PropTypes.number,
    }),
    sortBy: PropTypes.string,
    sortDirection: PropTypes.string,
    sortingData: PropTypes.object,
    isCollection: PropTypes.bool,
};

export default CommunityCollectionsSorting;
