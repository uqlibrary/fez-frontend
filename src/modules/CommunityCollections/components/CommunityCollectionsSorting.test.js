import React from 'react';
import CommunityCollectionsSorting from './CommunityCollectionsSorting';
// import { EXPORT_FORMAT_TO_EXTENSION } from 'config/general';

import { locale } from 'locale';

jest.mock('hooks');
// import { userIsAdmin } from 'hooks';
const txt = locale.components.communitiesCollections;
function setup(testProps = {}) {
    const props = {
        classes: {},
        pagingData: {
            from: 1,
            to: 20,
            total: 60,
            per_page: 20,
            current_page: 1,
        },
        sortingData: txt.sorting,
        canUseExport: false,
        location: {},
        disabled: false,
        activeFacets: { filters: {}, ranges: {} },
        onPageSizeChanged: jest.fn(),
        onSortByChanged: jest.fn(),
        onExportPublications: jest.fn(),
        ...testProps,
    };

    return getElement(CommunityCollectionsSorting, props);
}

describe('Community Collections component', () => {
    it('renders with empty paging data', () => {
        const data = {
            from: 0,
            to: 0,
            total: 0,
            current_page: 1,
        };
        const wrapper = setup({ pagingData: data });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders with non-empty community paging data', () => {
        const wrapper = setup();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders with export dropdown for admin or author', () => {
        const wrapper = setup({ canUseExport: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders with export dropdown hidden', () => {
        const wrapper = setup({ canUseExport: false });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders with all fields disabled', () => {
        const wrapper = setup({ disabled: true });
        wrapper.find('Select').forEach(option => {
            expect(option.props().disabled).toEqual(true);
        });
    });

    it('renders with non-empty paging data, pageChanged called', () => {
        const testFn = jest.fn();
        const wrapper = setup({ onPageSizeChanged: testFn });
        wrapper
            .find('#pageSize')
            .props()
            .onChange({ target: { value: 50 } });
        expect(testFn).toBeCalled();
    });

    it('renders with non-empty paging data, orderDirectionsChanged called', () => {
        const testFn = jest.fn();
        const testValue = 'test';
        const wrapper = setup({ onSortByChanged: testFn });
        wrapper
            .find('#sortOrder')
            .props()
            .onChange({ target: { value: testValue } });
        expect(testFn).toBeCalled();
    });

    it('renders with non-empty paging data, sortByChanged called', () => {
        const testFn = jest.fn();
        const testValue = 'test';
        const wrapper = setup({ onSortByChanged: testFn });
        wrapper
            .find('#sortBy')
            .props()
            .onChange({ target: { value: testValue } });
        expect(testFn).toBeCalled();
    });

    it('renders will set state on receiving new props', () => {
        const mockUseEffect = jest.spyOn(React, 'useEffect');
        const wrapper = setup({
            initPageLength: 5,
        });
        expect(toJson(wrapper)).toMatchSnapshot();

        mockUseEffect.mockImplementation(f => f());
        wrapper.setProps({
            sortBy: 'Publication date',
            sortDirection: 'test',
            pageSize: 5,
            pagingData: {},
        });

        expect(toJson(wrapper)).toMatchSnapshot();
        mockUseEffect.mockRestore();
    });
});
