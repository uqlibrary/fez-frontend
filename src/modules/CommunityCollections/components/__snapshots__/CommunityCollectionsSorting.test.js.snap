// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Community Collections component renders will set state on receiving new props 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Sort results by
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="sortBy"
        onChange={[Function]}
        value="published_date"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="0"
          value="title"
        >
          Title
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="1"
          value="created_date"
        >
          Created Date
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="2"
          value="updated_date"
        >
          Updated Date
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Sort order
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="sortOrder"
        onChange={[Function]}
        value="Desc"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="0"
          value="Desc"
        >
          Desc
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="1"
          value="Asc"
        >
          Asc
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Records per page
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="pageSize"
        onChange={[Function]}
        value={20}
      >
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-5"
          value={5}
        >
          5
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-10"
          value={10}
        >
          10
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-20"
          value={20}
        >
          20
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-50"
          value={50}
        >
          50
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-100"
          value={100}
        >
          100
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`Community Collections component renders will set state on receiving new props 2`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Sort results by
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="sortBy"
        onChange={[Function]}
        value="Publication date"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="0"
          value="title"
        >
          Title
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="1"
          value="created_date"
        >
          Created Date
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="2"
          value="updated_date"
        >
          Updated Date
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Sort order
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="sortOrder"
        onChange={[Function]}
        value="test"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="0"
          value="Desc"
        >
          Desc
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="1"
          value="Asc"
        >
          Asc
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Records per page
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="pageSize"
        onChange={[Function]}
        value={5}
      >
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-5"
          value={5}
        >
          5
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-10"
          value={10}
        >
          10
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-20"
          value={20}
        >
          20
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-50"
          value={50}
        >
          50
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-100"
          value={100}
        >
          100
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`Community Collections component renders with empty paging data 1`] = `
<span
  className="publicationsListSorting empty"
/>
`;

exports[`Community Collections component renders with export dropdown for admin or author 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Sort results by
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="sortBy"
        onChange={[Function]}
        value="published_date"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="0"
          value="title"
        >
          Title
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="1"
          value="created_date"
        >
          Created Date
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="2"
          value="updated_date"
        >
          Updated Date
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Sort order
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="sortOrder"
        onChange={[Function]}
        value="Desc"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="0"
          value="Desc"
        >
          Desc
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="1"
          value="Asc"
        >
          Asc
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Records per page
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="pageSize"
        onChange={[Function]}
        value={20}
      >
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-10"
          value={10}
        >
          10
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-20"
          value={20}
        >
          20
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-50"
          value={50}
        >
          50
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-100"
          value={100}
        >
          100
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`Community Collections component renders with export dropdown hidden 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Sort results by
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="sortBy"
        onChange={[Function]}
        value="published_date"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="0"
          value="title"
        >
          Title
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="1"
          value="created_date"
        >
          Created Date
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="2"
          value="updated_date"
        >
          Updated Date
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Sort order
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="sortOrder"
        onChange={[Function]}
        value="Desc"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="0"
          value="Desc"
        >
          Desc
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="1"
          value="Asc"
        >
          Asc
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Records per page
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="pageSize"
        onChange={[Function]}
        value={20}
      >
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-10"
          value={10}
        >
          10
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-20"
          value={20}
        >
          20
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-50"
          value={50}
        >
          50
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-100"
          value={100}
        >
          100
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`Community Collections component renders with non-empty community paging data 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Sort results by
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="sortBy"
        onChange={[Function]}
        value="published_date"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="0"
          value="title"
        >
          Title
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="1"
          value="created_date"
        >
          Created Date
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="2"
          value="updated_date"
        >
          Updated Date
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Sort order
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="sortOrder"
        onChange={[Function]}
        value="Desc"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="0"
          value="Desc"
        >
          Desc
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="1"
          value="Asc"
        >
          Asc
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
    xs={12}
  >
    <WithStyles(ForwardRef(FormControl))
      fullWidth={true}
    >
      <WithStyles(ForwardRef(InputLabel))
        shrink={true}
      >
        Records per page
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        disabled={false}
        id="pageSize"
        onChange={[Function]}
        value={20}
      >
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-10"
          value={10}
        >
          10
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-20"
          value={20}
        >
          20
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-50"
          value={50}
        >
          50
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="records-per-page-100"
          value={100}
        >
          100
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
